src/ado2gh/Commands/DownloadLogsCommand.cs

```diff
@@ -127,21 +136,36 @@ HttpDownloadService httpDownloadService
                 _log.LogWarning($"Overwriting {migrationLogFile} due to --overwrite option.");
             }
 
-            _log.LogInformation($"Downloading log for repository {githubRepo} to {migrationLogFile}...");
-
             var githubApi = _githubApiFactory.Create(githubApiUrl, githubPat);
-            var logUrl = await githubApi.GetMigrationLogUrl(githubOrg, githubRepo);
+            string logUrl;
 
-            if (logUrl == null)
+            while (true)
             {
-                throw new OctoshiftCliException($"Migration not found for repository {githubRepo}!");
-            }
+                logUrl = await githubApi.GetMigrationLogUrl(githubOrg, githubRepo);
 
-            if (!logUrl.HasValue())
-            {
-                throw new OctoshiftCliException($"Migration found for repository {githubRepo}, but migration log not available yet!");
+                if (logUrl.HasValue())
+                {
+                    break;
+                }
+
+                if (logUrl == null)
+                {
+                    throw new OctoshiftCliException($"Migration not found for repository {githubRepo}!");
```

Comment:
nit: IMO the `!` at the end of the error message might be confusing (might be considered as part of the repo name) so I'd just drop it.

Reply:
I updated the grammar instead in 7e402a2af69542e9241d83b4076787b2339d5f2f :slightly_smiling_face: How's this look?

https://github.com/github/gh-gei/blob/7e402a2af69542e9241d83b4076787b2339d5f2f/src/gei/Commands/DownloadLogsCommand.cs#L141-L141